Nos/Service/CurrentUser.swift

```diff
@@ -279,34 +281,72 @@ enum CurrentUserError: Error {
             website: author.website,
             picture: author.profilePhotoURL?.absoluteString
         ).dictionary
-        
         if let rawData = author.rawMetadata {
-            // Tack on any unsupported fields back onto the dictionary before publish
-            let rawJson = try? JSONSerialization.jsonObject(with: rawData)
-            if let rawJson, let rawDictionary = rawJson as? [String: AnyObject] {
-                for key in rawDictionary.keys {
-                    if metaEvent[key] == nil, let rawValue = rawDictionary[key] as? String {
-                        metaEvent[key] = rawValue
-                        Log.debug("Added \(key) : \(rawValue)")
+            // Tack on any unsupported fields back onto the dictionary before
+            // publish.
+            do {
+                let rawJson = try JSONSerialization.jsonObject(with: rawData)
+                if let rawDictionary = rawJson as? [String: AnyObject] {
+                    for key in rawDictionary.keys {
+                        guard metaEvent[key] == nil else {
+                            continue
+                        }
+                        if let rawValue = rawDictionary[key] as? String {
+                            metaEvent[key] = rawValue
+                            Log.debug("Added \(key) : \(rawValue)")
+                        }
                     }
                 }
+            } catch {
+                Log.debug("Couldn't parse a JSON from the user raw metadata")
+                // Continue with the metaEvent object we built previously
             }
         }
+        return metaEvent
+    }
 
-        let metaData = try? JSONSerialization.data(withJSONObject: metaEvent)
-        guard let metaData, let metaString = String(data: metaData, encoding: .utf8) else {
-            Log.debug("Error: Invalid meta data")
-            return
+    @MainActor func publishMetaData() async throws {
```

Comment:
```suggestion
    @MainActor func publishMetadata() async throws {
```

Metadata is one word, so let's spell it this way here and throughout. We do sometimes but haven't been consistent, so I'm hoping we can make progress in that direction.